R/snapshot.R

```diff
@@ -39,6 +41,41 @@ sd_snapshot <- function(self, private, items, filename, screenshot)
   url <- private$getTestSnapshotUrl(items$input, items$output, items$export)
   req <- httr::GET(url)
 
+  # Convert to text, then replace base64-encoded images with hashes of them.
+  content <- raw_to_utf8(req$content)
+  content <- hash_snapshot_image_data(content)
+
+  if(req$status_code != 200)
+  {
+    if(stop_on_error)
+    {
+      stop("Status code ", req$status_code, "\n", content)
+    } else {
+      new_content <- list(
+        status_code=req$status_code,
+        html=xml2::as_list(xml2::read_html(content))
+      )
+      content <- jsonlite::toJSON(new_content)
+    }
+  }
+
+
+  # Remove any items specified in ignore
```

Comment:
Implementing the `omit` arguments only involves this code block and adding the `omit` argument to the `sd_snapshot` and `app$snapshot` function definitions and handoff.